speech/google/cloud/speech/sample.py

```diff
@@ -51,17 +55,19 @@ class Sample(object):
     default_encoding = Encoding.FLAC
     default_sample_rate = 16000
 
-    def __init__(self, content=None, source_uri=None,
+    def __init__(self, content=None, source_uri=None, stream=None,
                  encoding=None, sample_rate=None, client=None):
         self._client = client
 
-        no_source = content is None and source_uri is None
-        both_source = content is not None and source_uri is not None
-        if no_source or both_source:
-            raise ValueError('Supply one of \'content\' or \'source_uri\'')
+        sources = [content is not None, source_uri is not None,
+                   stream is not None]
+        if sources.count(True) != 1:
```

Comment:
@daspecster My assumption was that creating the list is slower. However, I was wrong:

```
$ python -m timeit "A = B = None; C = 1; N = [A is not None, B is not None, C is not None].count(True)"
10000000 loops, best of 3: 0.182 usec per loop
$ python -m timeit "A = B = None; C = 1; N = int(A is not None) + int(B is not None) + int(C is not None)"
1000000 loops, best of 3: 0.306 usec per loop
```